### PR TITLE
hardening/364_mysql_connections_pool

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue https://github.com/telefonicaid/fiware-cygnus/issues/364
* 100% unit tests passed:
```
Results :
Tests run: 49, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed:
```
2015-05-11T17:55:39.205CEST | lvl=INFO | trans=1431359731-55-0000000000 | function=persist | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMySQLSink[225] : [mysql-sink] Persisting data at OrionMySQLSink. Database: z_serv, Table: null_other_rooms, Data: 1431359737,2015-05-11T15:55:37.542Z,Room1,Room,temperature,Room,26.5,[]
2015-05-11T17:55:39.205CEST | lvl=DEBUG | trans=1431359731-55-0000000000 | function=getConnection | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.mysql.MySQLBackend[243] : Connecting to jdbc:mysql://130.206.81.225:3306/z_serv?user=cb&password=XXXXXXXXXX
2015-05-11T17:59:48.387CEST | lvl=INFO | trans=1431359981-314-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[190] : Finishing transaction (1431359981-314-0000000000)
...
2015-05-11T17:59:55.419CEST | lvl=DEBUG | trans=1431359981-314-0000000001 | function=insertContextData | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.mysql.MySQLBackend[161] : Executing MySQL query 'insert into `null_other_rooms` values ('1431359993', '2015-05-11T15:59:53.267Z', 'Room1', 'Room', 'temperature', 'centigrade', '26.5', '[]')'
2015-05-11T17:59:55.480CEST | lvl=INFO | trans=1431359981-314-0000000001 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[190] : Finishing transaction (1431359981-314-0000000001)
...
mysql> select * from null_other_rooms;
+------------+--------------------------+----------+------------+-------------+------------+-----------+--------+
| recvTimeTs | recvTime                 | entityId | entityType | attrName    | attrType   | attrValue | attrMd |
+------------+--------------------------+----------+------------+-------------+------------+-----------+--------+
| 1431359986 | 2015-05-11T15:59:46.193Z | Room1    | Room       | temperature | centigrade | 26.5      | []     |
| 1431359993 | 2015-05-11T15:59:53.267Z | Room1    | Room       | temperature | centigrade | 26.5      | []     |
+------------+--------------------------+----------+------------+-------------+------------+-----------+--------+
```
* Assgnee @gtorodelvalle , but LGTM from @fgalan is also required.

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1 +1,2 @@
-OrionHDFSSink parameters are now called as "hdfs_*" instead of "cosmos_*" (#374)
+[FEATURE] OrionHDFSSink parameters are now called as "hdfs_*" instead of "cosmos_*" (#374)
```
Comment (gtorodelvalle) [inline CHANGES_NEXT_RELEASE]: Not important but it sounds better to me: '... are now called "hdfs_*"...' :) But I leave it to your discretion ;)

**Comment by gtorodelvalle** [comment]
LGTM on my side apart from a minor comment ;) Thanks @frbattid ! ;)

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/c99ceb6385615cdb40b6d3db92a0d8fbb1214d0f Thanks!

**Comment by fgalan** [comment]
LGTM

**Pull request merged**